packages/x-scheduler/src/internals/components/resource-legend/ResourceLegend.tsx

```diff
@@ -58,6 +58,31 @@ function ResourceLegendItem(props: { resource: CalendarResource }) {
   );
 }
 
+const getVisibilityDifferenceMap = (
```

Comment:
We discussed this behavior a few days ago.
I don't think you need to propagate, an event should be hidden if the parent of its resource is hidden, even if it's resource is not marked as hidden. Otherwise we can have broken state.

I think the cleanest option is to introduce a new utility function that uses `state.visibleResources` to check the resource and all its ancestors.

Reply:
That's exactly what I am proposing yes

To get the parent id from the children id, you can create a memoized selector that builds a lookup where the keys are the child id and the value the parent id.
We'll see later if we need to optimize that :+1: